apple-codesign/src/macho_signing.rs

```diff
@@ -143,6 +143,7 @@ fn create_macho_with_signature(
                     Ok(SEG_LINKEDIT) => {
                         let mut segment = *segment;
                         segment.filesize = new_linkedit_segment_size as _;
+                        segment.vmsize = new_linkedit_segment_size as _;
```

Comment:
This change (and the similar one a few lines later) seems unrelated to this PR. Would you mind splitting it into its own PR?

Reply:
Upon investigating this closer, there appears to be some nuance as to how `vmsize` is set. Specifically, it looks like the vmsize is rounded up to 8k boundaries or something along those lines.

I'm going to dig more and likely handle this in my own commit.